### ISSUE TITLE
Interior point method sometimes fails when given small variances and uniform mixcompdist.

### ISSUE THREAD

**Issue by dcgerard** [stephens999/ashr#17]
Example dataset:
``` r
betahat <- c(0.7775, 0.6091, 0.6880, 0.6897, 0.6565, 0.7505,
             0.7125, 0.7201, 0.7498, 0.7553)
sebetahat <- rep(0.01, length = length(betahat))
ip_out <- ashr::ash(betahat = betahat, sebetahat = sebetahat,
                    mixcompdist = "uniform", optmethod = "mixIP")
ip_out$PosteriorMean
min(ip_out$fitted.g$pi)
```

EM works fine:
``` r
em_out <- ashr::ash(betahat = betahat, sebetahat = sebetahat,
                    mixcompdist = "uniform", optmethod = "mixEM")
em_out$PosteriorMean
summary(em_out$fitted.g$pi)
```

**Comment by stephens999** [comment]
I've tracked this down further - the values are coming back from REBayes::KWDual.
Here is the simplest version I found. I emailed the REBayes maintainer to see if he
has insights.

 B=cbind(c(0,2e-9),c(.7,1))
REBayes::KWDual(B,rep(1,2),rep(1/2,2),control=list(verb=5))
$f
[1] -0.07218236  1.05345169

$g
[1] 0.7374162 1.0534517

$status
[1] "OPTIMAL"

**Comment by stephens999** [comment]
Roger Koenker responded: On your example, I would simply say that as in other optimization
settings one has to be careful about convergence tolerances, especially
when setting them to be greater than the resolution of the underlying
data.  When I do:

REBayes::KWDual(B,rep(1,2),rep(1/2,2),verb=5, rtol = 1e-10)

I get:

$f
[1] 6.074053e-07 9.999972e-01

$g
[1] 0.6999980 0.9999972

$status
[1] “OPTIMAL”

**Comment by stephens999** [comment]
addressed by ebe06a5
